hamilton/function_modifiers/delayed.py

```diff
@@ -148,3 +148,8 @@ def resolve(self, config: Dict[str, Any], fn: Callable) -> NodeTransformLifecycl
             if key in config:
                 kwargs[key] = config[key]
         return self.decorate_with(**kwargs)
+
+
+class resolve_from_config(resolve):
```

Comment:
Thought from @zilto  -- would `@resolve_at_config` be cleaner?

Also, let's add a docstring then stick it in the decorators documentation.

See https://github.com/DAGWorks-Inc/hamilton/blob/main/docs/reference/decorators/resolve.rst -- should be easy!

Reply:
I think `@resolve_at_config` better capture that it's a subset of `@resolve` where we set the value of `ResolveAt` to `ConfigAvailable`.

There's no other `ResolveAt` value, but I think the semantic make sense. I believe the name `@resolve_at_config` is an improvement over `@resolve` as it better communicate it's role during the Driver building.

Reply:
🤔  just to voice my opinion that `resolve_from_config` doesn't need you to understand what the driver does, but it does help more clearly tie the connection that it comes from configuration passed in. Which I think for a new person to Hamilton that might be more intuitive, than `resolve_at_config` since now they need to understand what "at" might be referring to.

@janhurst thoughts on the name?